addons/stock_account/models/stock.py

```diff
@@ -520,16 +520,21 @@ def _fifo_vacuum(self):
                     move.with_context(force_valuation_amount=corrected_value, forced_quantity=0)._account_entry_move()
 
     @api.model
-    def _run_fifo_vacuum(self):
+    def _run_fifo_vacuum(self, company_id=False):
         # Call `_fifo_vacuum` on concerned moves
-        fifo_valued_products = self.env['product.product']
-        fifo_valued_products |= self.env['product.template'].search([('property_cost_method', '=', 'fifo')]).mapped(
-            'product_variant_ids')
-        fifo_valued_categories = self.env['product.category'].search([('property_cost_method', '=', 'fifo')])
-        fifo_valued_products |= self.env['product.product'].search([('categ_id', 'child_of', fifo_valued_categories.ids)])
-        moves_to_vacuum = self.search(
-            [('product_id', 'in', fifo_valued_products.ids), ('remaining_qty', '<', 0)] + self._get_all_base_domain())
-        moves_to_vacuum._fifo_vacuum()
+        if company_id:
+            companies = [company_id]
+        else:
+            companies = self.env['res.company'].search([])
+        for company in companies:
+            fifo_valued_products = self.env['product.product']
+            fifo_valued_products |= self.env['product.template'].search([('property_cost_method', '=', 'fifo')]).mapped(
+                'product_variant_ids')
+            fifo_valued_categories = self.env['product.category'].search([('property_cost_method', '=', 'fifo')])
+            fifo_valued_products |= self.env['product.product'].search([('categ_id', 'child_of', fifo_valued_categories.ids)])
+            moves_to_vacuum = self.search(
+                [('product_id', 'in', fifo_valued_products.ids), ('remaining_qty', '<', 1)] + self._get_all_base_domain(company))
+            moves_to_vacuum._fifo_vacuum()
```

Comment:
No need to loop on all searches:
```suggestion
            fifo_valued_products = self.env['product.product']
            fifo_valued_products |= self.env['product.template'].search([('property_cost_method', '=', 'fifo')]).mapped(
                'product_variant_ids')
            fifo_valued_categories = self.env['product.category'].search([('property_cost_method', '=', 'fifo')])
            fifo_valued_products |= self.env['product.product'].search([('categ_id', 'child_of', fifo_valued_categories.ids)])
        for company in companies:
            moves_to_vacuum = self.search(
                [('product_id', 'in', fifo_valued_products.ids), ('remaining_qty', '<', 1)] + self._get_all_base_domain(company))
            moves_to_vacuum._fifo_vacuum()
```